File: BackBarbearia/Models/Barbeiro.cs

using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;


namespace Salao.Models
{
    public class Barbeiro : Pessoa
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id{get;set;}
        public int Unidadeid{get;set;}
        public UnidadeAtendimento? Unidade{get;set;}
    }
}

File: FrontBarbearia/src/app/Barbeiro.ts
import { Pessoa } from "./Pessoa";
import { unidadeAtendimento } from "./unidadeAtendimento";

export class Barbeiro extends Pessoa{
    id: number;
    unidadeId : number = 0;
    unidade : unidadeAtendimento | undefined;
    constructor(){
        super();
        this.id=0;
    }
}